### PR TITLE
docs: enhance local dev instructions

### LINKED ISSUE #149: PostgreSQL version in contributing.md no longer available.
**Describe the bug**
Installation instructions specify v14.7, however download site only offers v14.9 or 15.4. I installed 14.9. Let me know if this is the correct choice.

**To Reproduce**

1. Go to 'https://www.enterprisedb.com/downloads/postgres-postgresql-downloads'
2. Observe that v14.7 is not available

**Expected behavior**
Expected to find v14.7

**Screenshots**

**Environment (please complete the following information):**
Dev Environment

**Additional context**
At some point we may want to update contributing.md document.

### LINKED ISSUE #150: Update to DATABASE_URL gets tracked
**Describe the bug**
1. Noticed that db.md states to update DATABASE_URL in file 'packages/api/.env.local', however this file does not exist. It is called .env instead.
2. Noticed that .gitignore file has .env.local after the # Next.js heading.

**To Reproduce**
1. Created fork
3. Cloned to local environment
4. Installed PostgreSQL
5. Create local database
6. Updated DATABASE_URL in file: ./packages/api/.env
7. Git marks .env file as changed.

**Expected behavior**
1. Expected to find file env.local
2. Expected that when made update to DATABASE_URL that the file would be ignored by git.

**Screenshots**

**Environment (please complete the following information):**
- Device: Laptop
- OS: Windows 11

**Additional context**
Should .gitignore be updated to contain */packages/api/.env*  instead of *.env.local*?

### LINKED ISSUE #152: nx: command not found
**Describe the bug**
Was following instructions in `db.md`. Got a `nx: command not found` error when trying to run:
``` bash
nx prisma db migrate reset
```

**To Reproduce**
1. Fork repo
2. Install node.js
3. Install postgresql
4. Create local database
5. Set DATABASE_URL
6. Run in bash: `nx prisma db migrate reset`

**Expected behavior**
Expect command to run or instructions on how to configure local env to enable it to run.

**Screenshots**

**Environment (please complete the following information):**
- Laptop
- OS: Windows 11

**Additional context**
I discovered the answer to my problem in the `contributing.md` document.  You can close this issue. `nx` works now!

**Suggestions**
Perhaps we can reorder the sequence of steps in `contributing.md` and `db.md` to avoid the confusion I ran into.
Example:
1. Put these lines ***before*** the instructions for db install:
``` 
Run `npm install` to install all of the project's dependencies. 
Run `npm i -g nx` to make the `nx` command available in the terminal.
```
2.  Modify this line as follows and move ***before*** db install instruction but ***after*** above 2 lines.
```
Copy of `.env` file and rename it  `.env.local` file in the `api` package directory. 
Setup`env.local` as required. Instructions for setting the mandatory env var: `DATABASE_URL` are found in file `db.md`.
```
3. Move this line to the end of the `db.md` file:
```
Run `pg_restore -Fc --format=custom --dbname=<connection-string> data/seed.dump` to populate your database with development data. This includes all of the original language data as well as English glosses.
```

### LINKED ISSUE #173: How to logon as ADMIN in web app
**Describe the bug**
I have installed prostgres database and have seeded the db, but I don't have ability to login as ADMIN to test or use app.

**To Reproduce**
1. Install local env as per `contributing.md` and `db.md`.
2. Try to login. Not able to log into local web app as don't have valid email and pwd.

**Expected behavior**
Expected documentation to explain how to create a test ADMIN id.

**Environment (please complete the following information):**
- Laptop
- Windows 11

**Additional context**
Should this be added to `db.md`?

### PR DESCRIPTION
**Pull request by ilearner777**
## Connected Issues

closes #150,
closes #152,
closes #173

### Review file contributing.md

1. Open file contributing.md
2. Observe changes in the `Local Development` section as detailed below:
3. Added link to install `IDE VS Code` (clarify which IDE is being used for this project).
4. Add link to create a `GitHub` account. Help people who have never been exposed to GitHub.
5. Added PR workflow comments and tutorial. Help people who have never been exposed to GitHub.
6. Removed version number of `npm` as it changes with each release and is not significant to project.
7. Moved `npm install` instruction before the database instructions to load dependencies before db install.
8. Moved `npm i -g nx` instruction before the database instructions to avoid bash `nx not found` error during db install.
9. Added comment to db link to return to this page after db setup so don't forget to come back and complete this page.
10. Added links to web port:4300 and port:4200 for convenience.
11. Added ADMIN login credentials to web app.

### Review file db.md

1. Open file `db.md`.
2. Observe change in `Database`, `Setup` and `Seed and Reset Data` sections as detailed below:
3. Fixed link to `Database schema` file.
4. Removed `postgresql` minor version number as it changes overtime and is backward compatible.
5. Add hint on how to create a new database.
6. Clarified that `.env.local` is a copy of `packages/api/.env`.
7. Clarified that `packages/db/.env.local` is a new file to be created.
8. Added a `DATABASE_URL` template for convenience.
9. Replaced `nx seed db` with `pg_restore...` command in order to include user data in database so that developer will be able to log into the web app.
10. Added `language` to `code blocks` to remove lint errors and better format bash commands.

## Post-Deployment

- [ ] Nothing required

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -14,12 +14,15 @@ These are the major technologies we are using. Familiarity with most of these wi
...
-1. [Install](https://www.postgresql.org/download/) posgresql 14.7 and [set up a new database](./db.md) before running this project.
-1. Run `npm install` to install all of the project's dependencies. Run `npm i -g nx` to make the `nx` command available in the terminal.
-1. Set up your `.env.local` file in the `api` package. At minimum you need to set `DATABASE_URL` to a postgresql connection string to the database you created earlier.
-1. Run `pg_restore -Fc --format=custom --dbname=<connection-string> data/seed.dump` to populate your database with development data. This includes all of the original language data as well as English glosses.
-1. Run `nx serve api` and `nx serve web` in separate terminals. This will make the api server available on port 4300 and the web server available on 4200.
+1. [Install](https://code.visualstudio.com/) `IDE VS Code`.
+1. [Create](https://github.com/) a `GitHub` account (if you do not already have one).
+1. Create a `folk` on this repository. If you are new to GitHub you can work through this [tutorial](https://github.com/workdone0/first-contribution-github).
```
Comment (arrocke) [inline docs/contributing.md]: We don't need to specify an IDE. You can use whatever you want.

You also don't need a GitHub account or a fork to work with the code locally. These Local Development steps are meant to take your local code and get it running on your local machine. Instead, can we add a bullet to the Pull Requests section that explains the need for a fork before creating a PR?

File: docs/db.md [hunk 2]
```diff
@@ -2,53 +2,59 @@
...
 
-1. Install postgresql version 14.7
-1. Create database for the project
-1. Set the `DATABASE_URL` env var in `packages/api/.env.local` and `packages/db/.env.local` to the postgres connection string. You may need to include a username and password in the connection string.
-1. Run `nx prisma db migrate reset` to scaffold the database schema.
-1. Run `nx seed db` to insert static verse data into the database.
+1. [Install](https://www.postgresql.org/download/) postgresql version 14 (latest version).
+1. Create database for the project. You can perform this task using the included `pgAdmin` tool.
+1. Create a copy of file: [packages/api/.env](./../packages/api/.env) in same directory and rename it to: `.env.local`.
+1. Create new file: `packages/db/.env.local`.
```
Comment (arrocke) [inline docs/db.md]: You don't need to copy the .env file, you can just create `.env.local` in both files. Our build system loads both files. `.env.local` will override what is in `.env`. That's why we only need the DATABASE_URL in that file because it is unique to each machine.

Let's combine these two steps to say something like `Create a .env.local file in the api and db projects for local environment variables.`

File: docs/db.md [hunk 3]
```diff
@@ -2,53 +2,59 @@
...
+
+1. In a terminal run: `nx prisma db migrate reset` to scaffold the database schema.
+1. In a terminal run: `pg_restore -Fc --format=custom --dbname=<connection-string> data/seed.dump` to populate your database with development data. This includes all of the original language data as well as English glosses.
 
 ## Seed and Reset Data
 
 In order to reset the database to the current schema and clear all data, run the following command. It may take a few minutes to remove any existing data and build out the database schema.
 
-```
+```text
```
Comment (arrocke) [inline docs/db.md]:
```suggestion
```bash
```

File: packages/api/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "api",
+  "version": "0.0.1",
+  "dependencies": {
```
Comment (arrocke) [inline packages/api/package.json]: This package.json can be removed. We only need it in the root directory.

File: package-lock.json
Comment (arrocke) [inline package-lock.json:1]: Let's revert the changes to this file as well if our dependencies didn't actually change.

**Review by arrocke — CHANGES_REQUESTED** [review]
For future PRs, QA steps are where you specify what steps a reviewer should take on the app to test that everything behaves as expected. The "What has Changed" section is where you describe the highlights of what was changed in the PR. You don't need to describe everything because that would be the same as looking at the diff.

File: docs/contributing.md (hunk 1 above)
Comment (ilearner777) [reply docs/contributing.md]: I will remove the first 2 lines.
However, I was thinking of modifying the 3nd line to:

```
If you have never contributed to an open source repository before, you may find this [tutorial](https://github.com/workdone0/first-contribution-github) helpful.
```
and then putting this at the end of the `Your first contribution` section as the tutorial explains the process of performing one's first contribution, and includes the need to fork a project.

Let me know if you agree with change.

Comment (arrocke) [reply docs/contributing.md]: Yes! I like it

**Review by arrocke — CHANGES_REQUESTED** [review]
There are two unresolved comments from the previous review that need addressed yet.

**Review by arrocke — APPROVED** [review]
This looks good! Thanks
